pkg/controller/workload/workload_processor.go

```diff
@@ -467,16 +470,26 @@ func (p *Processor) handleDataWithoutService(workload *workloadapi.Workload) err
 }
 
 func (p *Processor) handleWorkload(workload *workloadapi.Workload) error {
-	log.Debugf("workload uid: %s", workload.Uid)
-	exist := cache.WorkloadCache.GetWorkloadByUid(workload.Uid)
-	cache.WorkloadCache.AddWorkload(workload)
-
-	// The grpc connection is disconnected every half hour and the workload will be resubscribed.
-	// We need to determine whether it already exists in the cache.
-	if exist == nil && isManagedWorkload(workload.Uid) {
-		Identity := GetIdentityByUid(workload.Uid)
-		p.Sm.SendCertRequest(Identity, kmeshsecurity.ADD)
+	log.Debugf("handle workload: %s", workload.Uid)
+	if p.isManagedWorkload(workload) {
+		oldIdentity := p.getIdentityByUid(workload.Uid)
+		newIdentity := spiffe.Identity{
+			TrustDomain:    workload.TrustDomain,
+			Namespace:      workload.Namespace,
+			ServiceAccount: workload.ServiceAccount,
+		}.String()
+		if oldIdentity == "" {
+			// This is the case workload added first time
+			p.Sm.SendCertRequest(newIdentity, kmeshsecurity.ADD)
+		} else if oldIdentity != newIdentity {
```

Comment:
maybe no need to check this, in real case workload identity cannot change unless recreate a pod